wails-frontend/src/pages/importPrivateKey.tsx

```diff
@@ -1,97 +1,131 @@
+import { IErrorObject, parseForm } from '../utils';
+import {
+  IsAlreadyExists,
+  IsNicknameValid,
+} from '../../wailsjs/go/walletapp/WalletApp';
+
+// TODO: create i18n and move this to translation file
+const t = (key: string): string => {
+  const errors: Record<string, string> = {
+    'nickname-required': 'Nickname is required',
+    'nickname-already-exists': 'This account name already exists',
+    'nickname-invalid-format':
+      "The account name can't contain any special characters",
+  };
+
+  return errors[key] || key;
+};
 
 const ImportPrivatekey = () => {
   const navigate = useNavigate();
+  const form = useRef(null);
 
   const [input, setInput] = useState<string | undefined>(undefined);
   const [privateKey, setPkey] = useState<string | undefined>(undefined);
+  const [error, setError] = useState<IErrorObject | null>(null);
 
   let { state } = useLocation();
   const req: promptRequest = state.req;
 
   const applyStr = 'Next';
 
-  const baselineStr = () => {
-    if (privateKey) {
-      return 'Choose an account name';
+  const baselineStr = () =>
+    privateKey ? 'Define your account name' : 'Enter your Private Key';
+
+  const placeholder = () => (privateKey ? 'Account name' : 'Private key');
+
+  async function validateNickname(e: SyntheticEvent) {
+    const formObject = parseForm(e);
+    const { nickname } = formObject;
+
+    if (!nickname) {
+      setError({ nickname: t('nickname-required') });
+      return false;
     }
-    return 'Enter your Private Key';
-  };
 
-  const placeholder = () => {
-    if (privateKey) {
-      return 'Account name';
+    if (await IsAlreadyExists(nickname)) {
+      setError({ nickname: t('nickname-already-exists') });
+      return false;
     }
-    return 'Private key';
-  };
 
-  const handleApply = () => {
-    if (!input || !input.length) {
-      return;
+    const nicknameIsValid = await IsNicknameValid(nickname);
+    if (!nicknameIsValid) {
+      setError({ nickname: t('nickname-invalid-format') });
+      return false;
     }
 
+    return true;
+  }
+
+  async function handleSubmit(e: SyntheticEvent) {
     if (!privateKey) {
-      // TODO: validate private key
+      // if the private key is not defined, set the private key
       setPkey(input);
+      // and reset the input
       setInput('');
-      return;
-    }
-
-    state = { ...state, pkey: privateKey, nickname: input };
+    } else {
+      const nicknameIsValid = await validateNickname(e);
 
-    navigate('/new-password', { state });
-  };
+      // if the private key is defined, validate the nickname
+      if (!nicknameIsValid) return;
+      // if the nickname is valid, navigate to the password page
+      state = { ...state, pkey: privateKey, nickname: input };
+      navigate('/new-password', { state });
+    }
+  }
 
-  const updateInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+  function updateInput(e: React.ChangeEvent<HTMLInputElement>) {
     setInput(e.target.value);
-  };
+  }
```

Comment:
Non blocking: Since we are doing nothing here and just wrapping a function maybe it's better to setup there!? 🤔 

```tsx
onChange={(e) => setInput(e.target.value)}
```

Reply:
but I use 2 time this function updateInput. I don't want to duplicate `onChange={(e) => setInput(e.target.value)}`